taca/illumina/__init__.py

```diff
@@ -0,0 +1,172 @@
+            cl.append('--with-failed-reads')
+        if cl_options.get('write-fastq-reverse-complement'):
+            cl.append('--write-fastq-reverse-complement')
+
+        LOG.info(("BCL to FASTQ conversion and demultiplexing started for "
+                  " run {} on {}".format(os.path.basename(run), datetime.now())))
+
+        misc.call_external_command_detached(cl, with_log_files=True)
+
+        LOG.info(("BCL to FASTQ conversion and demultiplexing finished for "
+                  "run {} on {}".format(os.path.basename(run), datetime.now())))
+
+
+def demultiplex_HiSeq(run):
+    """ Demultiplexing for HiSeq (V3/V4) runs
+    """
+    raise NotImplementedError('Meec! Demultiplexing for HiSeq (V3/V4) runs not implemented yet :-/')
+
+
+def demultiplex_MiSeq(run):
+    """ Demultiplexing for MiSeq runs
+    """
+    raise NotImplementedError('Meec! Demultiplexing for MiSeq runs not implemented yet :-/')
+
+
+class Run(object):
+    """ Defines an Illumina run
+    """
+    def __init__(self, run_dir):
+        if not os.path.exists(run_dir) or not \
+                os.path.exists(os.path.join(run_dir, 'runParameters.xml')):
+            raise RuntimeError('Could not locate run directory {}'.format(run_dir))
+        self.run_dir = run_dir
+        self.id = os.path.basename(run_dir)
+        self._extract_run_info()
+
+        if self.run_type == 'HiSeq':
+            self.demultiplex = demultiplex_HiSeq
+        elif self.run_type == 'HiSeq X':
+            self.demultiplex = demultiplex_HiSeq_X
+        elif self.demultiplex == 'MiSeq':
+            self.demultiplex = demultiplex_MiSeq
+
+
+    def _extract_run_info(self):
+        """ Extracts run info from runParameters.xml and adds it to the class attributes
+
+        TODO: This method could be used to extract A LOT of information about the
+        run and maybe... populate statusdb or similar? Just leaving this comment here
+        till now...
+        """
+
+        run_parameters = ET.parse(os.path.join(self.run_dir, 'runParameters.xml')).getroot().find('Setup')
+        # HiSeq and HiSeq X runParameter.xml files will have a Flowcell child with run type info
+        run_type = run_parameters.find('Flowcell')
+        # But MiSeqs doesn't...
+        if run_type is None:
+            run_type = run_parameters.find('ApplicationName')
+
+        try:
+            if 'HiSeq X' in run_type.text:
+                self.run_type = 'HiSeq X'
+            elif 'HiSeq Flow Cell' in run_type.text:
+                self.run_type = 'HiSeq'
+            elif 'MiSeq' in run_type.text:
+                self.run_type = 'MiSeq'
+        except AttributeError:
+            raise RuntimeError('Run type could not be determined for run {}'.format(self.id))
+
+
+    def is_finished(self):
+        """ Returns true if the run is finished, false otherwise
+        """
+        return os.path.exists(os.path.join(self.run_dir, 'RTAComplete.txt'))
+
+    @property
```

Comment:
haha,. @mariogiov that makes us two, I am reading upon that as well :joy: 

@guillermo-carrasco : you clever fox :smiling_imp: